sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
             });
         });
     },
+    
+    onDemonstrationDelete: function(demonstration, operation) {
+        var me = this,
+            dashboardView = me.view,
+            mainGridEl = dashboardView.el.down('.cbl-grid-main'),
+            studentId = demonstration.get('StudentID'),
+            competenciesStore = Ext.getStore('cbl-competencies-loaded'),
+            skillStudentsTpl = dashboardView.getTpl('skillStudentsTpl'),
+
+            deletedSkills = demonstration.get('Skills'),
+            deletedSkillsLength = deletedSkills.length, deletedSkillIndex = 0, demonstratedSkill,
+            loadedCompetency, loadedSkill, demonstrationsByStudent, skillDemonstrationsCell, deletedSkillIds, studentDemonstrations, studentDemonstrationIndex = 0,
+
+            competencyCompletions = operation.getProxy().getReader().rawData.data[0].competencyCompletions,
+            competencyCompletionsLength = competencyCompletions.length, competencyCompletionIndex = 0, competencyCompletion,
+            competencyProgressCell, competencyPercent,
+            _compentencyDemonstrationResetFn,
+            _updateCompetencySkillsFn, 
+
+            updatesCompetencies = [],
+            updatesLength, updateIndex, update, skillsRows, skillHeadersRow, skillStudentsRow, competencySkills, skill,
+            
+            updatedDemonstrationLength, updatedDemonstrationIndex;
+
+
+        // compile operations and DOM references for needed updates -- don't write to dom!
```

Comment:
Best practices in ExtJS appear to diverge significantly from JavaScript
best practices. It reminds me of Java.
On Feb 25, 2015 12:22 AM, "Chris Alfano" <notifications@github.com> wrote:

> In
> sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js
> <https://github.com/SlateFoundation/slate-cbl/pull/10#discussion_r25318721>
> :
>
> > +            deletedSkillsLength = deletedSkills.length, deletedSkillIndex = 0, demonstratedSkill,
> > +            loadedCompetency, loadedSkill, demonstrationsByStudent, skillDemonstrationsCell, deletedSkillIds, studentDemonstrations, studentDemonstrationIndex = 0,
> > +
> > +            competencyCompletions = operation.getProxy().getReader().rawData.data[0].competencyCompletions,
> > +            competencyCompletionsLength = competencyCompletions.length, competencyCompletionIndex = 0, competencyCompletion,
> > +            competencyProgressCell, competencyPercent,
> > +            _compentencyDemonstrationResetFn,
> > +            _updateCompetencySkillsFn,
> > +
> > +            updatesCompetencies = [],
> > +            updatesLength, updateIndex, update, skillsRows, skillHeadersRow, skillStudentsRow, competencySkills, skill,
> > +
> > +            updatedDemonstrationLength, updatedDemonstrationIndex;
> > +
> > +
> > +        // compile operations and DOM references for needed updates -- don't write to dom!
>
> What sort of context are you looking for? This is just the general best
> practice that heavy DOM manipulation should be batched into read and write
> phases. This is the read phase. Bouncing between reads and writes triggers
> reflows.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SlateFoundation/slate-cbl/pull/10/files#r25318721>.
>